cpp/src/arrow/acero/asof_join_node.cc

```diff
@@ -708,6 +747,21 @@ class InputState : public util::SerialSequencingQueue::Processor {
     auto rb = *batch.ToRecordBatch(schema_);
     DEBUG_SYNC(node_, "received batch from input ", index_, ":", DEBUG_MANIP(std::endl),
                rb->ToString(), DEBUG_MANIP(std::endl));
+    for (col_index_t i = 0; i < rb->num_columns(); i++) {
```

Comment:
I would also consider factoring this out to another function to keep the high level logic in `Process` still concise.